contracts/lsm-share-bond-provider/src/contract.rs

```diff
@@ -105,14 +105,24 @@ pub fn query(deps: Deps<NeutronQuery>, env: Env, msg: QueryMsg) -> ContractResul
             response: LAST_PUPPETEER_RESPONSE.may_load(deps.storage)?,
         })
         .map_err(From::from),
-        QueryMsg::TxState {} => query_tx_state(deps, env),
+        QueryMsg::TxState {} => query_tx_state(deps),
         QueryMsg::AsyncTokensAmount {} => {
             to_json_binary(&TOTAL_LSM_SHARES_REAL_AMOUNT.load(deps.storage)?).map_err(From::from)
         }
+        QueryMsg::CanBeRemoved {} => query_can_be_removed(deps, env),
     }
 }
 
-fn query_tx_state(deps: Deps<NeutronQuery>, _env: Env) -> ContractResult<Binary> {
+fn query_can_be_removed(deps: Deps<NeutronQuery>, env: Env) -> ContractResult<Binary> {
+    let all_balances = deps.querier.query_all_balances(env.contract.address)?;
+    let all_balances_except_untrn = all_balances
+        .into_iter()
+        .filter(|coin| coin.denom != *LOCAL_DENOM.to_string())
+        .collect::<Vec<Coin>>();
```

Comment:
Unfortunately this is not enough, we also need to be sure there is no any PENDING_LSM_SHARES and  LSM_SHARES_TO_REDEEM, and only after that we are safe to remove provider (during one block)